### DISCUSSION TITLE
Unable to import survey with 3.6 pre release

### DISCUSSION THREAD

**Discussion by vansorat** [ccbogel/QualCoder#962] [Q&A] [answered]
I can open a survey file in QC v. 3.5 but cannot do this in the mastercode (pre 3.6 release).  Here is the error message. Is anyone else having this problem.

  File "C:\Users\tuq68878\Downloads\QualCoder-master\qualcoder\__main__.py", line 1628, in import_survey
    ui = DialogImportSurvey(self.app, self.ui.textEdit)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\tuq68878\Downloads\QualCoder-master\qualcoder\import_survey.py", line 100, in __init__
    self.select_file()

  File "C:\Users\tuq68878\Downloads\QualCoder-master\qualcoder\import_survey.py", line 129, in select_file
    copyfile(self.filepath, destination)

  File "C:\Users\tuq68878\AppData\Local\Programs\Python\Python311\Lib\shutil.py", line 258, in copyfile
    with open(dst, 'wb') as fdst:
         ^^^^^^^^^^^^^^^

FileNotFoundError: [Errno 2] No such file or directory: 'C:/documents/TU SW Print and Electronic for qc 1 10.2024.csv'

**Comment by ccbogel** [comment]
Thank you for letting me know.
It is fixed now.

**Reply by vansorat** [reply]
Thank you for fixing that.  There seems to be a limit on the number of rows that can be imported with surveys222. I don't know what this is, but I tested various numbers.  I am able to import 6000 rows.  I also tested in NVivo and ATLAS.ti and those programs accepted less than 6000 rows (around 3000 was best).

With 8000 rows and anything more I get this error:

.  File "C:\Users\tuq68878\Downloads\QualCoder-master\qualcoder\__main__.py", line 1629, in import_survey
    ui = DialogImportSurvey(self.app, self.ui.textEdit)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\tuq68878\Downloads\QualCoder-master\qualcoder\import_survey.py", line 103, in __init__
    self.prepare_fields()

  File "C:\Users\tuq68878\Downloads\QualCoder-master\qualcoder\import_survey.py", line 241, in prepare_fields
    float(self.data[row][field])
          ~~~~~~~~~~~~~~^^^^^^^

IndexError: list index out of range

**Reply by ccbogel** [reply]
Hi, I am glad the earlier error is fixed, and that QC can load more rows than the other programs.
You are importing a csv file. It's probably that the CSV is not well formed around line 8000, I am guessing. As it is expecting a field and there is none. 
I Think load the csv with Excel and then save in excel format before loading into QC would be one option.
I will put in a work around, for a CSV format.

**Comment by vansorat** [comment]
Able to import csv with 3.6.  Does not recognize numeric and only gives me the option to change to qualitative.  My CSV has empty cells, some numeric columns and two qualitative columns.  Not sure I understand NONE, Minimal or ALL, but with NONE setting, empty cells show "missing value".  I selected delimiter comma.  With MINIMAL and NONE settings, the rows and columns look like my CSV sheet.  
The Unique Constraint warning dialog box is so long I cannot read the bottom of it. I just hit enter, because I have done this, otherwise, I would not know what to do.  This may be problematic for some users.  Then I get the import survey and OK popup and click OK.

**Reply by ccbogel** [reply]
I think generally, it is best to import using Excel.
CSV Comma Separated Values, the quotes around each datum can be for all, for none, or all except numbers:
None:   colin, 55, male
Minimal: "colin", 55, "male"
All: "Colin", "55, "male"

I have added the above info to the Wiki.

OK, Numeric, I might have to review how that is calculated, but, if every piece of data in the column is numeric, without text and without spaces or nothing, QC will view it as numeric

Ok I could edit how the unique constraint warning is displayed in the future

**Reply by ccbogel** [reply, marked as answer]
For QC 3.7 - the warning is now shorter and gives a count, not the case names details.  The numeric column now remains numeric if there are blank cell entries.
